core/src/main/java/org/elasticsearch/indices/IndicesService.java

```diff
@@ -1073,4 +1118,13 @@ public void onRemoval(RemovalNotification<IndicesRequestCache.Key, IndicesReques
 
     }
 
+    @FunctionalInterface
+    interface IndexDeletionPredicate extends BiFunction<Index, IndexSettings, Boolean> {
+        Boolean apply(Index index, IndexSettings indexSettings);
```

Comment:
Is the boxing necessary? Can we just not extend `BiFunction` and use a primitive `boolean`?